packages/@aws-cdk/aws-elasticsearch/lib/elasticsearch-access-policy.ts

```diff
@@ -20,6 +21,14 @@ export interface ElasticsearchAccessPolicyProps {
    * The access policy statements for the Elasticsearch cluster
    */
   readonly accessPolicies: iam.PolicyStatement[];
+
+  /**
+   * The key used for cluster encryption at rest.
+   * Needed in order to grant kms permissions on this key for the operation to succeed.
+   *
+   * @default - no key.
+   */
+  readonly kmsKey?: kms.IKey;
```

Comment:
In what scenario are the permissions required? Should we add a default key creation is this use case? Or add a warning/error in case this is required? 

From the issue you have attached and from the documentation on this property it read like this is a common pitfall

Reply:
The permissions are only required if the domain configures encryption at rest with a custom key. It's optional here because some domains won't.

I've refactored the code a bit to apply the permissions on the domain side instead of accepting it here.